game.cpp

```diff
@@ -52,7 +53,32 @@ void game_setup(main_character &character) {
   generate_all_alterers();
   generate_all_monsters();
   set_initial_items();
-  character.game_progress_multiplier = 1;
+  character.game_progress_multiplier = DIFFICULTY_LEVEL;
+}
+
+void select_difficulty_level() {
+  printf("%s", DIFFICULTY_OPTIONS.c_str());
+  int option;
+  bool valid_option = true;
```

Comment:
this setting would allow to enter in the "do" of the do-while? Its while not valid_option, and if valid_option is true at the beginning, it would start as do { ... } while { not true }

Reply:
in a do while statement the code inside the do's scope executes at least once, then, the expression inside the while is checked, if it returns true, the code inside the do's scope executes again..